diff -ruN -x 'cscope.*' splash2-orig/codes/kernels/cholesky/block2.C splash2-bounded/codes/kernels/cholesky/block2.C
--- splash2-orig/codes/kernels/cholesky/block2.C	2010-05-05 00:36:21.000000000 -0400
+++ splash2-bounded/codes/kernels/cholesky/block2.C	2011-08-08 18:11:29.000000000 -0400
@@ -180,7 +180,7 @@
 	if (LB.renumbering[BLOCK(i)->i] < 0 ||
 	    LB.renumbering[BLOCK(i)->j] < 0) {
 	  printf("Block %ld has bad structure\n", which);
-	  exit(-1);
+	  // exit(-1);
 	}
 	BLOCK(i)->done = 0;
 	BLOCK(i)->pair = NULL;
@@ -316,7 +316,7 @@
     LB.col[col+1] = LB.col[col] + n_nz - (col-super);
     if (LB.col[col+1] > LB.entries_allocated) {
       printf("Overflow\n");
-      exit(-1);
+      // exit(-1);
     }
 
     for (i=col-super; i<n_nz; i++)
@@ -351,7 +351,7 @@
 
   if (LB.col[LB.n+which_domain+1] > LB.entries_allocated) {
     printf("Overflow!!\n");
-    exit(-1);
+    // exit(-1);
   }
 }
 
@@ -406,7 +406,7 @@
   LB.col[super+1] = LB.col[super] + n_nz;
   if (LB.col[super+1] > LB.entries_allocated) {
     printf("Overflow\n");
-    exit(-1);
+    // exit(-1);
   }
   for (i=0; i<n_nz; i++)
     LB.row[LB.col[super]+i] = nz[i];
diff -ruN -x 'cscope.*' splash2-orig/codes/kernels/cholesky/malloc.C splash2-bounded/codes/kernels/cholesky/malloc.C
--- splash2-orig/codes/kernels/cholesky/malloc.C	2010-05-05 00:36:21.000000000 -0400
+++ splash2-bounded/codes/kernels/cholesky/malloc.C	2011-08-08 18:30:27.000000000 -0400
@@ -282,11 +282,11 @@
 
   if (size <= 0) {
     printf("Bad size %ld\n", size);
-    exit(-1);
+    // exit(-1);
   }
   if (home < -1 || home >= mallocP) {
     printf("Bad home %ld\n", home);
-    exit(-1);
+    // exit(-1);
   }
 
   if (size > MAXFASTBL)
diff -ruN -x 'cscope.*' splash2-orig/codes/kernels/cholesky/numLL.C splash2-bounded/codes/kernels/cholesky/numLL.C
--- splash2-orig/codes/kernels/cholesky/numLL.C	2010-05-05 00:36:21.000000000 -0400
+++ splash2-bounded/codes/kernels/cholesky/numLL.C	2011-08-08 18:12:01.000000000 -0400
@@ -234,7 +234,7 @@
   diag = *theNZ;
   if (diag <= 0.0) {
     printf("Negative pivot, d[%ld] = %f\n", j, diag);
-    exit(0);
+    // exit(0);
   }
   diag = sqrt(diag);
   *theNZ++ = diag;
diff -ruN -x 'cscope.*' splash2-orig/codes/kernels/cholesky/util.C splash2-bounded/codes/kernels/cholesky/util.C
--- splash2-orig/codes/kernels/cholesky/util.C	2010-05-05 00:36:21.000000000 -0400
+++ splash2-bounded/codes/kernels/cholesky/util.C	2011-08-08 18:12:50.000000000 -0400
@@ -41,7 +41,7 @@
 
   if (!M.col || !M.row || (nz && !M.nz)) {
     printf("NewMatrix %ld %ld: Out of memory\n", n, m);
-    exit(0);
+    // exit(0);
   }
 
   return(M);
@@ -66,7 +66,7 @@
 
   if (!v && n) {
     printf("Out of memory: NewVector(%ld)\n", n);
-    exit(0);
+    // exit(0);
   }
 
   return(v);
@@ -121,7 +121,7 @@
 	type[3] = 0;
 	if (!(type[0] != 'C' && type[1] == 'S' && type[2] == 'A')) {
 	  fprintf(stderr, "Wrong type: %s\n", type);
-	  exit(0);
+	  // exit(0);
 	}
 
 	fscanf(fp, "%11c", buf); /* pad */
